### PR TITLE
Deprecate SafeERC20.safeIncreaseAllowance and SafeERC20.safeDecreaseAllowance

### PR DESCRIPTION
**Pull request by Amxx**
#### PR Checklist

- [ ] Tests
- [x] Documentation
- [x] Changeset entry (run `npx changeset add`)

### CODE REVIEW

**Review by arr00 — DISMISSED** [review]
LGTM

File: .changeset/sharp-rules-joke.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': minor
+---
+
+`SafeERC20`: deprecate `safeIncreaseAllowance` and `safeDecreaseAllowance` and documents risks when associated with ERC-7674.
```
Comment (cairoeth) [inline .changeset/sharp-rules-joke.md:5]:
```suggestion
`SafeERC20`: Deprecate `safeIncreaseAllowance` and `safeDecreaseAllowance`, and document risks when associated with ERC-7674.
```

**Review by cairoeth — COMMENTED** [review]
should `forceApprove` also have a comment?

**Comment by Amxx** [comment]
> should `forceApprove` also have a comment?

No deprecation, but a comment yes maybe

**Review by cairoeth — APPROVED** [review]

**Review by cairoeth — APPROVED** [review]
I'm not a big fan of marking `safeIncreaseAllowance` and `safeDecreaseAllowance` as deprecated given their [common usage](https://sourcegraph.com/search?q=context:global+repo:tintinweb/smart-contract-sanctuary+file:/mainnet/+.safeIncreaseAllowance%28&patternType=standard&sm=0) in production contracts, so I opened #5262, which is the same but without the deprecation. I'll still approve this in case you think deprecation still makes sense despite the usage.

**Pull request closed without merging**